src/wallet/scriptpubkeyman.cpp

```diff
@@ -1929,3 +1937,32 @@ bool DescriptorScriptPubKeyMan::AddCryptedKey(const CKeyID& key_id, const CPubKe
     m_map_crypted_keys[key_id] = make_pair(pubkey, crypted_key);
     return true;
 }
+
+bool DescriptorScriptPubKeyMan::HasWalletDescriptor(const WalletDescriptor& desc) const
+{
+    if (descriptor.descriptor != nullptr && desc.descriptor != nullptr && descriptor.descriptor->ToString() == desc.descriptor->ToString()) {
```

Comment:
Could just return this entire boolean expression rather than using an `if` statement.